lib/ssh/src/ssh_connection_handler.erl

```diff
@@ -121,11 +121,11 @@ start_link(Role, _Address=#address{}, Id, Socket, Options) ->
     end.
 
 
-takeover(ConnPid, client, Socket, Options) ->
+takeover(ParentPid, ConnPid, client, Socket, Options) ->
     group_leader(group_leader(), ConnPid),
-    takeover(ConnPid, common, Socket, Options);
+    takeover(ParentPid, ConnPid, common, Socket, Options);
 
-takeover(ConnPid, _, Socket, Options) ->
+takeover(ParentPid, ConnPid, _, Socket, Options) ->
```

Comment:
Here we will be changing  controlling process a second time. I think we should write the code in such a way that we only need to do it once. The acceptor process that should be  in the supervisor tree should create the socket and start  the appropriate subsystem dynamic supervisor tree and retrieve the pid of the connection_handler procsee and transfer the ownership once.